### PR TITLE
PBI 13 - Melihat Daftar Itinerary

### PR DESCRIPTION
**Pull request by fathanadhitama**
CHANGES:
- Developed the GET /itineraries/me endpoint to retrieve a user's itineraries.
- Implemented a robust pagination system to efficiently return an optimal list of itineraries.

NOTES:
- Development Approach:
Followed Test-Driven Development (TDD) to ensure reliable and well-tested code.
- Validation:
1. Returning lists in expected format if itineraries exist [POSITIVE CASE]
2. Returning an empty list if user has no itineraries [NEGATIVE CASE]
3. Throwing an Exception if queried page is invalid (page < 1 or page > totalPages) [CORNER CASE]
- Achieving 100% unit test coverage for both the Itinerary Controller and Itinerary Service.

ATTACHMENTS:
- Code coverage screenshot:
image

### CODE REVIEW

File: src/itinerary/itinerary.controller.ts [hunk 1]
```diff
@@ -25,6 +35,27 @@ export class ItineraryController {
     return this.itineraryService.findAll()
   }
 
+  @Public()
```
[the remark this answers is not in the document]
Comment (fathanadhitama) [reply src/itinerary/itinerary.controller.ts]: Terima kasih Nasya! Benar juga, penggunaan @Public() di endpoint ini memungkinkan akses oleh pengguna yang tidak seharusnya memiliki akses dan berpotensi menjadi isu keamanan yang penting. Saya sudah menghapus decorator @Public() untuk memastikan hanya pengguna yang terautentikasi yang dapat mengakses endpoint ini.
